### PR TITLE
Add http method, url, flavor and status_code attributes to client spans

### PR DESCRIPTION
**Pull request by mpichette-apple**
Fixes https://github.com/trace4cats/trace4cats-http4s/issues/54

Requires changes on trace4cats-model library!

### CODE REVIEW

File: modules/http4s-client/src/main/scala/io/janstenpickle/trace4cats/http4s/client/Http4sClientRequest.scala [hunk 1]
```diff
@@ -6,8 +6,15 @@ import io.janstenpickle.trace4cats.model.{AttributeValue, SemanticAttributeKeys}
 import org.http4s.Uri
 
 object Http4sClientRequest {
+  // FIXME: move to trace4cats-model...
+  val SemanticAttributeKeys_httpFlavor: String = "http.flavor"
```
Comment (mpichette-apple) [inline modules/http4s-client/src/main/scala/io/janstenpickle/trace4cats/http4s/client/Http4sClientRequest.scala:10]: I would be happy to move this into trace4cats-model if/once these changes are accepted.

Comment (mpichette-apple) [reply modules/http4s-client/src/main/scala/io/janstenpickle/trace4cats/http4s/client/Http4sClientRequest.scala:10]: preemptively created PR for it: https://github.com/trace4cats/trace4cats/pull/659

File: modules/http4s-client/src/main/scala/io/janstenpickle/trace4cats/http4s/client/Http4sClientResponse.scala [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import io.janstenpickle.trace4cats.http4s.common.Response_
+import io.janstenpickle.trace4cats.model.{AttributeValue, SemanticAttributeKeys}
+
+object Http4sClientResponse {
+
+  def toAttributes(res: Response_): Map[String, AttributeValue] = {
+    Map[String, AttributeValue](
+      SemanticAttributeKeys.httpStatusCode -> res.status.code,
+    )
+  }
```
Comment (catostrophe) [inline modules/http4s-client/src/main/scala/io/janstenpickle/trace4cats/http4s/client/Http4sClientResponse.scala:12]:
```suggestion
  def toAttributes(res: Response_): Map[String, AttributeValue] =
    Map[String, AttributeValue](
      SemanticAttributeKeys.httpStatusCode -> res.status.code,
    )
```

**Review by catostrophe — CHANGES_REQUESTED** [review]

**Review by catostrophe — APPROVED** [review]

File: modules/http4s-common/src/main/scala/io/janstenpickle/trace4cats/http4s/common/Http4sSpanNamer.scala [hunk 3]
```diff
@@ -9,6 +9,15 @@ object Http4sSpanNamer {
 
   val methodWithPath: Http4sSpanNamer = req => s"${req.method.name} ${Uri.decode(req.pathInfo.toString)}"
 
+  /** OpenTelemetry's recommended default for http clients
+    *
+    * See [[https://opentelemetry.io/docs/reference/specification/metrics/semantic_conventions/http-metrics/#parameterized-attributes]]
+    */
+  val httpMethod: Http4sSpanNamer = req => s"HTTP ${req.method.name}"
+
+  val spanNameAttrOrHttpMethod: Http4sSpanNamer = req =>
```
[the remark this answers is not in the document]
Comment (tarmath) [reply modules/http4s-common/src/main/scala/io/janstenpickle/trace4cats/http4s/common/Http4sSpanNamer.scala]: This scheme is to allow people to manually specify a span name when they create the request, for times when either the default span namers just don't capture the essence of it, but also when folks want to name their client spans to give it more local context.

[the remark this answers is not in the document]
Comment (catostrophe) [reply modules/http4s-common/src/main/scala/io/janstenpickle/trace4cats/http4s/common/Http4sSpanNamer.scala]: Sure, I don't mind adding more namers. I just wondered if you had a useful snippet demonstrating this approach.

[the remark this answers is not in the document]
Comment (catostrophe) [reply modules/http4s-common/src/main/scala/io/janstenpickle/trace4cats/http4s/common/Http4sSpanNamer.scala]: @tarmath what if change it to the following?
```scala
def spanNameAttrOrElse(fallback: Http4sSpanNamer): Http4sSpanNamer = req =>
  req.attributes.lookup(Http4sAttributes.Keys.SpanName).getOrElse(fallback(req))
```

[the remark this answers is not in the document]
Comment (tarmath) [reply modules/http4s-common/src/main/scala/io/janstenpickle/trace4cats/http4s/common/Http4sSpanNamer.scala]: ahhh, wish I'd thought of it myself!

[the remark this answers is not in the document]
Comment (catostrophe) [reply modules/http4s-common/src/main/scala/io/janstenpickle/trace4cats/http4s/common/Http4sSpanNamer.scala]: fine. I'll fix and merge

**Pull request merged**